sql.go

```diff
@@ -71,37 +74,44 @@ func (*SQL) Open(database string, connectionString string) (*dbsql.DB, error) {
 
 // Query executes the provided query string against the database, while
 // providing results as a slice of KeyValue instance(s) if available.
-func (*SQL) Query(db *dbsql.DB, query string, args ...interface{}) ([]keyValue, error) {
-	rows, err := db.Query(query, args...)
-	if err != nil {
-		return nil, err
+func (*SQL) Query(db *dbsql.DB, query string, args ...interface{}) ([]KeyValue, error) {
+	rows, _ := db.Query(query, args...)
+	defer func() {
+		_ = rows.Close()
+	}()
+	if rows.Err() != nil {
+		return nil, rows.Err()
 	}
+
 	cols, err := rows.Columns()
 	if err != nil {
 		return nil, err
 	}
+
 	values := make([]interface{}, len(cols))
 	valuePtrs := make([]interface{}, len(cols))
-	result := make([]keyValue, 0)
+	result := make([]KeyValue, 0)
 
 	for rows.Next() {
 		for i := range values {
 			valuePtrs[i] = &values[i]
 		}
 
-		err := rows.Scan(valuePtrs...)
-
+		err = rows.Scan(valuePtrs...)
 		if err != nil {
 			return nil, err
 		}
 
-		data := make(keyValue, len(cols))
+		data := make(KeyValue, len(cols))
 		for i, colName := range cols {
-			data[colName] = *valuePtrs[i].(*interface{})
+			valPtr, ok := valuePtrs[i].(*interface{})
+			if !ok {
+				return nil, fmt.Errorf("expected pointer type")
```

Comment:
This seems like something a linter would detect, but I don't see how a type assertion to `*interface{}` could ever fail(?).

A couple thoughts about this:

- Even if it does fail, we likely should panic and treat it as an urgent bug, and not like a normal application error. After all, what can the user do if they see "expected pointer type"? If it's not a panic with a scary stack trace, they could even miss the error altogether, which we definitely don't want.

- Now that we're building and testing with Go 1.20, we can use `any` instead of `interface{}`. Similarly, could generics help us avoid pointers here? Summoning our generics guru @oleiade for any thoughts on the matter.

If this is too much detail to dig into now, I vote for silencing the linter for now, and creating another issue where we can look into it in depth.

This extension compliance PR should only make things compatible with the recent registry requirements. Let's not make any functional changes and risk regressions because of it.